.github/workflows/promote.yml

```diff
@@ -0,0 +1,91 @@
+name: promote-contracts
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      environment:
+        required: true
+        type: choice
+        options:
+          - sbx
+          - prod_test
+          - prod
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  promote:
+    runs-on: self-hosted
+    steps:
+      - name: Set environment
+        id: set-env
+        run: echo "env_name=${{ inputs.environment }}" >> $GITHUB_OUTPUT
```

Comment:
Why introduce this level of abstraction instead of referencing inputs.environment directly?